users/signals.py

```diff
@@ -0,0 +1,27 @@
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+from .models import User, Profile, UserSettings
+from .helpers import nickname_from_email
+
+
+# Do I need other signals?
+
+@receiver(post_save, sender=User)
+def create_profile(sender, instance, created, **kwargs):
+    if created:
+        profile = Profile.objects.create()
+        if instance.profile and instance.nickname:
+            nickname = instance.profile.nickname
```

Comment:
Wow wait. If there already is instance.profile then there is nothing to do, no?
Especially that nickname has to be required upon registration, as on reddit. 
Reminder: it was inferred from email only for migration.

Reply:
No nickname inferation any more